### DISCUSSION TITLE
Saturation and Response Curves for multiple models created in hierarchical modelling

### DISCUSSION THREAD

**Discussion by chiranjib-acn-cda** [google/meridian#316] [Q&A]
We are trying to use Hierarchical Model for National level model and what to run several model for the various filters and KPI. Does Meridian allow optimization with different response and saturation curve for various models.

For example, for a particular country we want to find out marketing effectiveness on android, iOS and android + iOS (i.e., at an overall level). If I want to use hierarchical modelling, is it possible to get saturation and response curves for all 3 scenarios (pre optimization) and get different saturation and response curve for the optimization.

**Comment by yingxzgo** [comment]
To generate response curves by different KPIs (or their combinations), there are two possible approaches you can try:

**1.Hierarchical Model Approach:**
You can build a hierarchical model where each KPI group (e.g., Android, iOS) is treated as a "geo" in the load.CoordToColumns() function. For this, you'll need to segment both media spend and impressions by each group (KPI). Once that's done, you can use the "selected_geos" argument within the analyzer.response_curves()function to generate the response curve for each KPI or KPI combination.

**2. Separate National Models Approach:**
Alternatively, you can build separate national models for each KPI individually. Once you have the response curves for each KPI, you can combine them manually. The rationale here is that the expected value of the sum of KPIs is the sum of the individual expected values, i.e., E(y1+y2∣media)=E(y1∣media)+E(y2∣media). This method does not require segment media spend & impressions by each group (KPI).

However, budget optimization currently does not support geo slices within hierarchical models. This means that with approach 1, you won’t be able to get optimal budget allocation for each KPI (we might work on this in the future). Additionally, budget optimization does not currently work with manually created response curves. As a result, with approach 2, you won’t be able to determine the optimal budget allocation for KPI combinations.
